### PR TITLE
Feature/matrix load by reassignment

### PR DESCRIPTION
**Pull request by meirhalachmi**

### CODE REVIEW

File: src/corona_hakab_model/generation/matrix_generator.py [hunk 1]
```diff
@@ -62,63 +59,50 @@ def import_connection_data(import_file_path: str) -> "ConnectionData":
 
 
 class MatrixData:
-    __slots__ = ("matrix_type", "matrix", "depth")
```
Comment (bgmoshe) [inline src/corona_hakab_model/generation/matrix_generator.py:65]: Why you remove this?

File: src/corona_hakab_model/generation/matrix_generator.py [hunk 2]
```diff
@@ -62,63 +59,50 @@ def import_connection_data(import_file_path: str) -> "ConnectionData":
...
+    def __getstate__(self):
```
Comment (bgmoshe) [inline src/corona_hakab_model/generation/matrix_generator.py:62]: why not make this a property instead of function?

Comment (bgmoshe) [inline src/corona_hakab_model/generation/matrix_generator.py:62]: Why not implement _setstate_ ?

File: src/corona_hakab_model/generation/matrix_generator.py [hunk 3]
```diff
@@ -138,35 +122,23 @@ def __init__(
...
-                        con_type_data, self.social_circles_by_connection_type[con_type], current_depth
-                    )
-                current_depth += 1
-
-        # current patch since matrix is un-serializable
-        self.matrix_data.matrix_type = "parasymbolic"
-        self.matrix_data.matrix = self.matrix
-        self.matrix_data.depth = self.depth
+        # todo switch the depth logic, to get a connection type instead of int depth
+        for current_depth, (con_type, con_type_data) in enumerate(self.connection_types_data.items()):
```
Comment (bgmoshe) [inline src/corona_hakab_model/generation/matrix_generator.py:129]: Who you do not lock anymore?

File: src/corona_hakab_model/generation/matrix_generator.py [hunk 4]
```diff
@@ -213,13 +187,17 @@ def _create_fully_connected_circles_matrix(self, con_type_data: ConnectionTypeDa
             return
 
         for circle in circles:
+            if circle.agent_count <= 1:
```
Comment (bgmoshe) [inline src/corona_hakab_model/generation/matrix_generator.py:192]: You checked if this happens?

**Review by bgmoshe — COMMENTED** [review]
Overall good. Few qs needs to be answered.

File: src/corona_hakab_model/generation/matrix_generator.py (hunk 1 above)
Comment (meirhalachmi) [reply src/corona_hakab_model/generation/matrix_generator.py:65]: It interrupts with the pickling

File: src/corona_hakab_model/generation/matrix_generator.py (hunk 2 above)
Comment (meirhalachmi) [reply src/corona_hakab_model/generation/matrix_generator.py:62]: you don't need it. you need `getstate` for the pickling. I'll add it as a comment

File: src/corona_hakab_model/generation/matrix_generator.py (hunk 3 above)
Comment (meirhalachmi) [reply src/corona_hakab_model/generation/matrix_generator.py:129]: Cause you don't touch the matrix here, only the assignment array

File: src/corona_hakab_model/generation/matrix_generator.py (hunk 4 above)
Comment (meirhalachmi) [reply src/corona_hakab_model/generation/matrix_generator.py:192]: Yes, there are 1-agent circles

**Review by tal-nc — APPROVED** [review]

**Pull request merged**